spyrmsd/rmsd.py

```diff
@@ -373,3 +373,181 @@ def rmsdwrapper(
+    molref: molecule.Molecule,
+    mols: Union[molecule.Molecule, List[molecule.Molecule]],
+    queue: Queue,
+    symmetry: bool = True,
+    center: bool = False,
+    minimize: bool = False,
+    strip: bool = True,
+    cache: bool = True,
+) -> Any:
+    """
+    Compute RMSD between two molecules and put it in a queue.
+
+    Parameters
+    ----------
+    molref: molecule.Molecule
+        Reference molecule
+    mols: Union[molecule.Molecule, List[molecule.Molecule]]
+        Molecules to compare to reference molecule
+    queue: Queue
+        The queue where to put the result
+    symmetry: bool, optional
+        Symmetry-corrected RMSD (using graph isomorphism)
+    center: bool, optional
+        Center molecules at origin
+    minimize: bool, optional
+        Minimised RMSD (using the quaternion polynomial method)
+    strip: bool, optional
+        Strip hydrogen atoms
+
+    Returns
+    -------
+    None
+        The RMSD result is put in the queue.
+    """
+    queue.put(
+        rmsdwrapper(
+            molref=molref,
+            mols=mols,
+            symmetry=symmetry,
+            center=center,
+            minimize=minimize,
+            strip=strip,
+            cache=cache,
+        )
+    )
+
+
+def rmsd_timeout(
+    molref: molecule.Molecule,
+    mols: Union[molecule.Molecule, List[molecule.Molecule]],
+    symmetry: bool = True,
+    center: bool = False,
+    minimize: bool = False,
+    strip: bool = True,
+    cache: bool = True,
+    timeout: Optional[float] = 5,  ## Do you think this is a good default timeout value?
+) -> Any:
+    """
+    Compute RMSD between two molecules with a timeout.
+
+    Parameters
+    ----------
+    molref: molecule.Molecule
+        Reference molecule
+    mols: Union[molecule.Molecule, List[molecule.Molecule]]
+        Molecules to compare to reference molecule
+    symmetry: bool, optional
+        Symmetry-corrected RMSD (using graph isomorphism)
+    center: bool, optional
+        Center molecules at origin
+    minimize: bool, optional
+        Minimised RMSD (using the quaternion polynomial method)
+    strip: bool, optional
+        Strip hydrogen atoms
+    timeout: float, optional
+        After how many seconds to stop the RMSD calculations
+
+    Returns
+    -------
+    List[float]
+        RMSDs
+    """
+
+    if not isinstance(mols, list):
+        mols = [mols]
+
+    ## Inspired by https://superfastpython.com/task-with-timeout-child-process/
+    queue = Queue()
+    process = Process(
+        target=_rmsd_queue,
+        args=(molref, mols, queue, symmetry, center, minimize, strip, cache),
+    )
+
+    process.start()
+    process.join(timeout=timeout)
+
+    ## Check if the process finished running successfully
+    if not process.exitcode == 0:
+        ## Actually terminate the process
+        process.terminate()
+
+        ## Match the length of the mols list (Maybe this should be handled differently?)
+        return [None] * len(mols)
```

Comment:
```suggestion
        return [np.nan] * len(mols)
```
I'm wondering if `np.nan` would be better here, mainly because `np.nan` can be used in an non-object NumPy array:
```console
>>> import numpy as np
>>> np.asarray([None, None])
array([None, None], dtype=object)
>>> np.asarray([np.nan, np.nan])
array([nan, nan])
```
What do you think?

Reply:
Yeah, I wasn't sure what was the best way to handle this. `np.nan` is probably the best solution here, thanks!

Reply:
Instead of the list, I think this would be faster:
```
n = len(mols)
r = numpy.empty((n,n))
r[:] = np.nan
```